ovn-tester/ovn_workload.py

```diff
@@ -216,6 +196,123 @@ def wait(self, sbctl, timeout_s):
+        cluster.nbctl.ls_port_del(port)
+        self.unbind_port(port)
+        self.lports.remove(port)
+
+    @ovn_stats.timeit
+    def bind_port(self, port):
+        log.info(f'Binding lport {port.name} on {self.container}')
+        self.vsctl.add_port(port, 'br-int', internal=True, ifaceid=port.name)
+        # Skip creating a netns for "passive" ports, we won't be sending
+        # traffic on those.
+        if not port.passive:
+            self.vsctl.bind_vm_port(port)
+
+    @ovn_stats.timeit
+    def unbind_port(self, port):
+        if not port.passive:
+            self.vsctl.unbind_vm_port(port)
+        self.vsctl.del_port(port)
+
+    def provision_ports(self, cluster, n_ports, passive=False):
+        ports = [self.provision_port(cluster, passive) for i in range(n_ports)]
+        for port in ports:
+            self.bind_port(port)
+        return ports
+
+    def run_ping(self, cluster, src, dest):
+        log.info(f'Pinging from {src} to {dest}')
+
+        # FIXME
+        # iputils is inconsistent when working with sub-second timeouts.
+        # The behavior of ping's "-W" option changed a couple of times already.
+        # https://github.com/iputils/iputils/issues/290
+        # Until that's stable use "timeout 0.1s" instead.
+        cmd = f'ip netns exec {src} timeout 0.1s ping -q -c 1 {dest}'
+        start_time = datetime.now()
+        while True:
+            try:
+                self.run(cmd=cmd, raise_on_error=True)
+                break
+            except ovn_exceptions.SSHError:
+                pass
+
+            duration = (datetime.now() - start_time).seconds
+            if duration > cluster.cluster_cfg.node_timeout_s:
+                log.error(
+                    f'Timeout waiting for {src} ' f'to be able to ping {dest}'
+                )
+                raise ovn_exceptions.OvnPingTimeoutException()
+
+    @ovn_stats.timeit
+    def ping_port(self, cluster, port, dest):
+        self.run_ping(cluster, port.name, dest)
+
+    def ping_ports(self, cluster, ports):
+        for port in ports:
+            if port.ip:
+                self.ping_port(cluster, port, dest=port.ext_gw)
+            if port.ip6:
+                self.ping_port(cluster, port, dest=port.ext_gw6)
+
+    def get_connection_string(self, cluster_cfg, port):
+        protocol = "ssl" if cluster_cfg.enable_ssl else "tcp"
+        offset = 0
+        offset += 3 if cluster_cfg.clustered_db else 1
+        offset += cluster_cfg.n_relays
+        return f"{protocol}:{self.mgmt_ip + offset}:{port}"
+
+    def configure(self, physical_net):
+        raise NotImplemented
+
+    @ovn_stats.timeit
+    def provision(self, cluster):
+        raise NotImplemented
+
+    @ovn_stats.timeit
+    def provision_port(self, cluster, passive=False):
+        raise NotImplemented
+
+    @ovn_stats.timeit
+    def ping_external(self, cluster, port):
+        raise NotImplemented
+
+
+class WorkerNode(ChassisNode):
```

Comment:
@dceara it is this inheritance that made me think of `ChassisNode` as `Compute` is not really fit with Kubernetes. Or what do you think?

Reply:
It could also serve as a base for a `NetworkNode`

Reply:
I'm ok with both; if `ChassisNode` makes more sense to you let's go with that. :)